snuba/state/__init__.py

```diff
@@ -93,14 +100,20 @@ def wrapper() -> Any:
 
 
 def numeric(value: Any) -> Any:
+    # Return the given value based on its correct type
+    # It supports the following types: int, float, bool, string
     try:
-        assert isinstance(value, (str, SupportsInt))
+        assert isinstance(value, (str, int))
         return int(value)
     except (ValueError, AssertionError):
         try:
             assert isinstance(value, (str, SupportsFloat))
             return float(value)
         except (ValueError, AssertionError):
+            if value == "True" or value == "true":
+                return True
+            elif value == "False" or value == "false":
+                return False
```

Comment:
Added to support boolean's. In redis, booleans are stored as strings "True" or "False"